services/clusters-keeper/src/simcore_service_clusters_keeper/modules/clusters.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+import logging
+from typing import cast
+
+from fastapi import FastAPI
+from models_library.users import UserID
+from models_library.wallets import WalletID
+from servicelib.logging_utils import log_context
+from types_aiobotocore_ec2.literals import InstanceTypeType
+
+from ..core.errors import Ec2InstanceNotFoundError
+from ..core.settings import get_application_settings
+from ..models import EC2InstanceData
+from ..utils.clusters import create_startup_script
+from ..utils.ec2 import (
+    HEARTBEAT_TAG_KEY,
+    all_created_ec2_instances_filter,
+    creation_ec2_tags,
+    ec2_instances_for_user_wallet_filter,
+)
+from .ec2 import get_ec2_client
+
+_logger = logging.getLogger(__name__)
+
+
+async def create_cluster(
+    app: FastAPI, *, user_id: UserID, wallet_id: WalletID
+) -> list[EC2InstanceData]:
+    ec2_client = get_ec2_client(app)
+    app_settings = get_application_settings(app)
+    assert app_settings.CLUSTERS_KEEPER_EC2_INSTANCES  # nosec
+    return await ec2_client.start_aws_instance(
+        app_settings.CLUSTERS_KEEPER_EC2_INSTANCES,
+        instance_type=cast(
+            InstanceTypeType,
+            next(
+                iter(
+                    app_settings.CLUSTERS_KEEPER_EC2_INSTANCES.EC2_INSTANCES_ALLOWED_TYPES
+                )
```

Comment:
Q: Later here we will inject the instance type chosen by the user?

Reply:
this is still work in progress. this one machine is basically where the dask-gateway runs, so I am not sure it needs a super machine.